samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -0,0 +1,254 @@
+/* Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include <modem/location.h>
+#include <net/nrf_cloud.h>
+#include "nrf_cloud_codec.h"
+#include <stdio.h>
+#include <date_time.h>
+#include <cJSON.h>
+#include <math.h>
+
+#include "application.h"
+#include "temperature.h"
+#include "connection.h"
+
+#include "location_tracking.h"
+
+LOG_MODULE_REGISTER(application, CONFIG_MQTT_MULTI_SERVICE_LOG_LEVEL);
+
+/* Timer used to time the sensor sampling rate. */
+K_TIMER_DEFINE(sensor_sample_timer, NULL, NULL);
+
+
+
+/**
+ * @brief Construct a data device message cJSON object with automatically generated timestamp.
+ *
+ * @param appid
+ * @return cJSON* - the timestamped data device message object if successful, NULL otherwise.
+ */
+static cJSON *create_timestamped_data_message_object(const char *const appid)
+{
+	cJSON *msg_obj = NULL;
+	int64_t timestamp;
+
+	if (date_time_now(&timestamp)) {
+		LOG_ERR("Failed to create timestamp for data message "
+			"with appid %s", log_strdup(appid));
+		return NULL;
+	}
+
+	msg_obj = cJSON_CreateObject();
```

Comment:
This can return `NULL` if there's no memory available. The code should probably check that before using `msg_obj`.

Reply:
Good point, thank you, I will correct that!